File: youpinhui-sellergoods-interface/src/main/java/com/youpinhui/sellergoods/service/SpecificationService.java
package com.youpinhui.sellergoods.service;
import java.util.List;

import com.youpinhui.entity.PageResult;
import com.youpinhui.pojo.TbSpecification;
import com.youpinhui.pojogroup.Specification;

/**
 * provider  Specification interface
 * @author Leon
 *
 */
public interface SpecificationService {

	/**
	 * find all 
	 * @return
	 */
	public List<TbSpecification> findAll();
	
	
	/**
	 * query according pagination 
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * Add
	*/
	public void add(Specification specification);
	
	
	/**
	 * update
	 */
	public void update(Specification specification);
	

	/**
	 * get one by id
	 * @param id
	 * @return
	 */
	public Specification findOne(Long id);
	
	
	/**
	 * batch delete
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 
	 * fuzzy search 
	 *  if the object  is not null , get the parameter from it as the search condition.
	 *
	 * @param pageNum current page number
	 * @param pageSize size of the page to display
	 * @return
	 */
	public PageResult findPage(TbSpecification specification, int pageNum,int pageSize);
	
}

File: youpinhui-manager-web/src/main/webapp/js/service/specificationService.js
//Sevice
app.service('specificationService',function($http){
	    	
	 //test case , get all the data return a Json
	this.findAll=function(){
		return $http.get('../specification/findAll.do');		
	}
	//pagination
	this.findPage=function(page,rows){
		return $http.get('../specification/findPage.do?page='+page+'&rows='+rows);
	}
	//find one object
	this.findOne=function(id){
		alert("id : "+id);
		return $http.get('../specification/findOne.do?id='+id);
	}
	//add 
	this.add=function(entity){
		return  $http.post('../specification/add.do',entity );
	}
	//update 
	this.update=function(entity){
		return  $http.post('../specification/update.do',entity );
	}
	//batch delete
	this.dele=function(ids){
		return $http.get('../specification/delete.do?ids='+ids);
	}
	//search
	this.search=function(page,rows,searchEntity){
		return $http.post('../specification/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
